src/GaugeInput.js

```diff
@@ -0,0 +1,220 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import TextField from '@material-ui/core/TextField';
+import Snackbar from '@material-ui/core/Snackbar';
+import Alert from '@material-ui/lab/Alert';
+import ExpansionPanel from '@material-ui/core/ExpansionPanel';
+import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
+import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import Button from '@material-ui/core/Button'
+
+const api = require('./api.js');
+
+const styles = theme => ({
+  container: {
+    display: 'flex',
+    flexDirection: 'row',
+    width: '100%'
+  },
+  textField: {
+    width: '60%',
+    marginRight: 15
+  },
+  expansionPanelDetails: {
+    display: 'flex',
+    flexDirection: 'column'
+  }
+});
+
+class GaugeInput extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      id: `${this.props.id}`,
+      index: `${this.props.index}`,
+      prob: [],
+      success: false,
+      error: false,
+      warning: false,
+      gauge: this.props.gauge
+    }
+  }
+
+  handleOnChange = event => {
+    console.log('Click');
+    console.log(event.target.value);
+  };
+
+  open = (result) => {
+    console.log(result)
+    if (result.errors) {
+      if (this.props.pass === "") {
+        this.setState({warning: true})
+      } else {
+        this.setState({error: true})
+      }
+    } else {
+      this.setState({success: true})
+    }
+  }
+
+  close = (event, reason) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+    if (this.state.error) {
+      this.setState({error: false})
+    } else if (this.state.warning){
+      this.setState({warning: false})
+    } else {
+      this.setState({success: false})
+    }
+  }
+
+  render() {
+    const { classes } = this.props;
+    let { gauge, prob } = this.state;
+    let text = []
+
+    if (gauge.messages) {
+      gauge.messages.map((m => {
+        prob.push(m.probability)
+        text.push(m.text)
+      }))
+    }
+
+    const updateProb = event => {
+      prob[event.target.id][event.target.getAttribute('bin')] = event.target.value
+    }
+
+    const updateText = event => {
+      text[event.target.id] = event.target.value
+    }
+
+    const addMessage = () => {
+      api.post(`glyphs/${this.state.id}/gauges/${this.state.index}/messages`,
+        {"pass" : this.props.pass, "text" : 'Default message', "probability" : [0,0,0,0,0]}
+      ).then(result => {console.log(result)})
+
+      if (!gauge.messages) {
+        gauge[ "messages" ] = [];
+      }
+      gauge.messages.push({"text" : 'Default gauge message', "probability" : [0,0,0,0,0]})
+      this.forceUpdate();
```

Comment:
Once again, you will want to refactor this to use `this.setState` so that you don't have to force update.  It will get a little messy here because you are modifying a property of the state variable - see below snippet:

```
this.setState(prevState => ({
    jasper: {                   // object that we want to update
        ...prevState.jasper,    // keep all other key-value pairs
        name: 'something'       // update the value of specific key
    }
}))
```